### DISCUSSION TITLE
BadUnexpectedError on method call

### DISCUSSION THREAD

**Discussion by butzmeister** [FreeOpcUa/opcua-asyncio#1646] [Q&A] [answered]
Hi,
I'm getting an BadUnexpectedError when trying to call the 'node_1_illuminate_multiple_slots' method on 'art_net' using the code below.

This is my code:
```      
parent_id = 'ns=3;i=1'
child_id = 'ns=3;i=11'

parent_node = self._client.get_node(parent_id)
child_node = self._client.get_node(child_id)
    
converted_args = []
for arg in args:
        converted_args.append(ua.Variant(arg, ua.VariantType.String))

out = await parent_node.call_method(child_node, converted_args)
```

Setup on Server:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/62953892/fc5898d8-e973-4cd5-bf25-1a7e82ee52d6)

Parent Node:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/62953892/a8422e50-1bff-41f8-9369-e017c6cbb143)

Child Node:
![image](https://github.com/FreeOpcUa/opcua-asyncio/assets/62953892/34b35f70-114a-4ed2-bd19-30badd5f2b1a)

Calling the method with same arguments using node-RED and UaExpert works fine. 

I can access the Nodes to read the available methods or read the description, but calling the method does not.

Has anyone had similar issues and can help?
Thank you!

**Comment by AndreasHeine** [comment, marked as answer]
how does the method inputarguments look like? is it really an array?

small example: https://github.com/AndreasHeine/opcua-tutorial/blob/main/client%2Fcall_method.py

**Reply by butzmeister** [reply]
thank you very much for the hint, andreas. 
it's working now, I did not pass the input arguments correctly.

this is the working code:
```
parent_id = 'ns=3;i=1'
child_id = 'ns=3;i=11'

parent_node = self._client.get_node(parent_id)
child_node = self._client.get_node(child_id)
    
converted_args = []
for arg in args:
     converted_args.append(ua.Variant(arg, ua.VariantType.String))

out = await parent_node.call_method(child_node, converted_args[0], converted_args[1])
```
